controllers/cruisecontroloperation_controller.go

```diff
@@ -0,0 +1,445 @@
+				obj := e.Object.(*banzaiv1alpha1.CruiseControlOperation)
+				return obj.GetCurrentTaskOp() != ""
+			},
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				oldObj := e.ObjectOld.(*banzaiv1alpha1.CruiseControlOperation)
+				newObj := e.ObjectNew.(*banzaiv1alpha1.CruiseControlOperation)
+				if !reflect.DeepEqual(oldObj.GetCurrentTask(), newObj.GetCurrentTask()) ||
+					oldObj.GetDeletionTimestamp() != newObj.GetDeletionTimestamp() ||
+					oldObj.GetGeneration() != newObj.GetGeneration() {
+					return true
+				}
+				return false
+			},
+		})
+
+	return builder
+}
+
+func getKafkaClusterReference(operation *banzaiv1alpha1.CruiseControlOperation) (client.ObjectKey, error) {
+	if operation.GetClusterRef() == "" {
+		return client.ObjectKey{}, errors.NewWithDetails("could not find kafka cluster reference label for CruiseControlOperation", "missing label", banzaiv1beta1.KafkaCRLabelKey, "name", operation.GetName(), "namespace", operation.GetNamespace())
+	}
+	return client.ObjectKey{
+		Name:      operation.GetClusterRef(),
+		Namespace: operation.GetNamespace(),
+	}, nil
+}
+
+func updateResult(res *scale.Result, operation *banzaiv1alpha1.CruiseControlOperation, isAfterExecution bool) error {
+	if res == nil || operation.GetCurrentTask() == nil {
+		return nil
+	}
+	task := operation.GetCurrentTask()
+
+	if (task.State == banzaiv1beta1.CruiseControlTaskCompleted || task.State == banzaiv1beta1.CruiseControlTaskCompletedWithError) && task.Finished == nil {
+		task.Finished = &v1.Time{Time: time.Now()}
+	}
+
+	// Add the failed task into the status.failedTasks slice only when the update is happened after executing task
+	if isAfterExecution && task.Finished != nil && task.State == banzaiv1beta1.CruiseControlTaskCompletedWithError {
+		if len(operation.Status.FailedTasks) >= defaultFailedTasksHistoryMaxLength {
+			operation.Status.FailedTasks = append(operation.Status.FailedTasks[1:], *task)
+		} else {
+			operation.Status.FailedTasks = append(operation.Status.FailedTasks, *task)
+		}
+		operation.Status.NumberOfRetries += 1
+		operation.Status.ErrorPolicy = banzaiv1alpha1.ErrorPolicyRetry
+		task.Finished = nil
+		task.State = ""
+		task.Started = nil
+	}
+
+	task.ID = res.TaskID
+	if task.Started == nil {
+		//Sat, 27 Aug 2022 12:22:21 GMT
+		startTime, err := time.Parse(time.RFC1123, res.StartedAt)
+		if err != nil {
+			return errors.WrapIff(err, "could not parse user task start time from Cruise Control API")
+		}
+		task.Started = &v1.Time{Time: startTime}
+	}
+	// We only have summary after execution
+	if isAfterExecution {
+		task.Summary = parseSummary(res.Result)
+		task.ErrorMessage = res.Err
+	}
+	// This is necessary because of a bug in CC (when a user task resulted in error later when get usertasks status from CC the corresponding task state is Active and not CompletedWithError...)
+	//if task.State != banzaiv1beta1.CruiseControlTaskCompletedWithError && task.State != banzaiv1beta1.CruiseControlTaskCompleted {
```

Comment:
According to the comment here, the if check is needed, but it is currently commented out

Reply:
After I found that the problem caused by the get user task 100 limit what I have removed. 
https://github.com/banzaicloud/koperator/blob/b5df80b9dec3efc8ee1b6085af947ee074c56c87/pkg/scale/scale.go#L118
I will remove this part. If the problem does not come back.